### PR TITLE
DT-2341 Word Count API 

### PR DESCRIPTION
**Pull request by 2un**
[Word Count API - update old TMS word count endpoints to proxy requests to reports service](https://bt.smartling.net/browse/{DT-2341})
- feat (public SDK for word count report api)

### CODE REVIEW

File: smartling-reports-api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
```
Comment (kgavrilchenko) [inline smartling-reports-api/pom.xml:9]: why do we need compiler plugin in sub module?

Comment (2un) [reply smartling-reports-api/pom.xml]: removed

File: smartling-reports-api/src/main/java/com/smartling/api/reports/v3/ReportsApi.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+     *    fields: List of fields in the responce. IF empty - return list of columns by default.
+     *    includeTranslationResource: deprecated. Use fields instead.
+     *    includeJob: deprecated. Use fields instead.
+     *    includeJobReferenceNumber: deprecated. Use fields instead.
+     *    includeFuzzyMatchProfile: deprecated. Use fields instead.
+     *    includeWorkflowStep: deprecated. Use fields instead.
+     */
+    @POST
+    @Path("word-count/csv")
+    void downloadWordCountReport(WordCountReportCommandPTO wordCountReportCommandPTO);
```
Comment (DmitryBochorishvili) [inline smartling-reports-api/src/main/java/com/smartling/api/reports/v3/ReportsApi.java:48]: Does our sdk really support csv streaming endpoints?

File: smartling-reports-api/src/main/java/com/smartling/api/reports/v3/ReportsApi.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+@Path("/reports-api/v3")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public interface ReportsApi
+{
+    /**
+     * Obtains word count report data
+     * @param wordCountReportCommandPTO command with filters for the report
+     *    date fields using pattern: yyyy-MM-dd
```
Comment (DmitryBochorishvili) [inline smartling-reports-api/src/main/java/com/smartling/api/reports/v3/ReportsApi.java:21]:
```suggestion
     *    date fields using pattern: yyyy-MM-dd, treated as in America/New_York timezone
```

**Pull request closed without merging**